Create Blog “create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers”

GuopingJia opened the pull request:
Automatically generated by Netlify CMS

drensing commented on content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md:
```diff
@@ -0,0 +1,261 @@
+
+3. The kubeconfig is generated for the HPE Ezmeral Runtime Enterprise. It contains commands to show version, authenticate and refresh of the environment. The standard kubectl tool installed from the official Kubernetes site does not work with this kubeconfig file. You have to download the HPE kubectl plugin, available from the same dashboard, and use it together with the kubectl and the kubeconfi ...
+
+You need a cleaner solution to have a kubeconfig file that is not tied to a specific user, has the right set of privileges, and works permanently with the standard kubectl tool. This blog post walks you through the process of creating a general-purpose kubeconfig file that allows you to access and deploy applications to the Kubernetes cluster in HPE GreenLake for Containers. The created kubeconfi ...
```
Recommend modifying the last sentence to read:
"By using these instructions, you'll be able to create kubeconfig files that can be used by any external scripts specific to your CI/CD pipeline set up and have them work with the Kubernetes cluster."

drensing commented on content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md:
```diff
@@ -0,0 +1,261 @@
+You need a cleaner solution to have a kubeconfig file that is not tied to a specific user, has the right set of privileges, and works permanently with the standard kubectl tool. This blog post walks you through the process of creating a general-purpose kubeconfig file that allows you to access and deploy applications to the Kubernetes cluster in HPE GreenLake for Containers. The created kubeconfi ...
+
+## Prerequisites
+You need to download the kubectl binary, together with the HPE kubectl plugin and the kubeconfig file, from the launched HPE Ezmeral Runtime Enterprise Dashboard. The downloaded kubectl binary and its plugin need to be set up in your environment. To simplify the setup process, you export the environment variable `KUBECONFIG` and point it to the downloaded kubeconfig file. With those setups in pla ...
```
Change "Dashboard" to "dashboard" (lower case "d")
Change "you export the environment..." to "you should export the environment..."
Change "With those setups" to "With these setups"

drensing commented on content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md:
```diff
@@ -0,0 +1,261 @@
+## Prerequisites
+You need to download the kubectl binary, together with the HPE kubectl plugin and the kubeconfig file, from the launched HPE Ezmeral Runtime Enterprise Dashboard. The downloaded kubectl binary and its plugin need to be set up in your environment. To simplify the setup process, you export the environment variable `KUBECONFIG` and point it to the downloaded kubeconfig file. With those setups in pla ...
+
+You have access to permissions that can create and update the following resources in the Kubernetes cluster:
```
Would it make sense to modify this to "Now you have access to permissions..."? Or is this something that you had already?

drensing commented on content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md:
```diff
@@ -0,0 +1,261 @@
+
+## Setup Details
+### Create a Kubernetes Service Account
+You use the following yaml manifest file to create a service account in the Kubernetes cluster. Replace the name `cfe-demo-sa` with your service account name.
```
Recommend changing "You use" to "You can use"

drensing commented on content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md:
````diff
@@ -0,0 +1,261 @@
+cfe-demo-sa   1         24s
+```
+### Create a Role
+After you have the service account, you create a role with a set of permissions which represents the access rights that you want for your scripts. 
````
Change "you create a role" to "you can create a role"
Change "which" to "that" and "represents" to "represent"

drensing commented on content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md:
````diff
@@ -0,0 +1,261 @@
+```
+
+### Extract Service Account Token
+You check the secret token in the created service account and extract the token field by running the following commands. The token is a randomized string. The setup shows only a snippet of this string.
````
Change "You check the" to "You can now check the"

drensing commented on content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md:
```diff
@@ -0,0 +1,261 @@
+
+If you access the Kubernetes API directly, e.g., from `curl`, you can use the token as the bearer token for the authorization header. 
+
+However, if you have your scripts running outside the cluster that use kubectl or a client library to access the Kubernetes cluster, you need the kubeconfig file to load configs from. You follow up the following section to create such a kubeconfig file.
```
Change "cluster that use kubectl" to "cluster that uses kubectl"
Change "to load configs from" to "to load the configs from"
Change the last sentence to read "Follow the instructions in the next section to create such a kubeconfig file."

drensing commented on content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md:
````diff
@@ -0,0 +1,261 @@
+rm tmp.min
+
+```
+After running the script, a general-purpose kubeconfig file `kubeconfig-sa` is created. Export the kubeconfig file as the environment variable `KUBECONFIG`, you can access the Kubernetes cluster and check all the resources deployed in the cluster.
````
Change the beginning of the second sentence to read:
"By exporting the kubeconfig file as..."

drensing commented on content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md:
````diff
@@ -0,0 +1,261 @@
+No resources found in cfe-demo-cluster namespace.
+```
+## Conclusion
+This blog post shows you how to create a general-purpose kubeconfig file using a service account. The kubeconfig file is not tied to any specific user. It is bound with a list of permissions carefully chosen for your need to access the Kubernetes cluster. The created kubeconfig file works permanently with both downloaded kubectl binary from HPE GreenLake for Containers Dashboard and the standard  ...
````
Modify the beginning of the second sentence to:
"This kubeconfig file is not tied..."
In the second to last sentence add "the" between "both" and "downloaded kubectl"

drensing reviewed (commented):
Jia,
I think if you can clarify one or two things in this post at the beginning, this will be an amazing post! I've made some suggestions on what it is that I THINK you mean. If you want to talk about it, please feel free to reach out.

GuopingJia commented:
> Jia, I think if you can clarify one or two things in this post at the beginning, this will be an amazing post! I've made some suggestions on what it is that I THINK you mean. If you want to talk about it, please feel free to reach out.

Hi, Dale, thank you for your comments and suggestions. As always, I will carefully walk through all the points and get them addressed. 
HPE GreenLake for containers service, same as HPE GreenLake for private cloud (GLPC) service, is provided as part of HPE GreenLake Central (GLC) portal. You can click the service widget to get its main 'Clusters' page, then start creating your cluster from this portal. HPE Ezmeral Runtime Enterprise is acting as a backend. Though all the clusters will be created in HPE Ezmeral, it's NOT visible to customers in GLC portal. After your cluster gets created, you can "launch" to the HPE Ezmeral Runtime Enterprise, landing to its dashboard. The kubectl, together with the HPE kubectl plugin and its kubeconfig, is all from this cross-launched dashboard. 
I will try to make this more clear in the Introduction page. 

Guoping

GuopingJia commented:
> > Jia, I think if you can clarify one or two things in this post at the beginning, this will be an amazing post! I've made some suggestions on what it is that I THINK you mean. If you want to talk about it, please feel free to reach out.
> 
> Hi, Dale, thank you for your comments and suggestions. As always, I will carefully walk through all the points and get them addressed. HPE GreenLake for containers service, same as HPE GreenLake for private cloud (GLPC) service, is provided as part of HPE GreenLake Central (GLC) portal. You can click the service widget to get its main 'Clusters' page, then start creating your cluster from this portal. HPE Ezmeral Runtime Enterprise is acting as a backend. Though all the clusters will be created in HPE Ezmeral, it's NOT visible to customers in GLC portal. After your cluster gets created, you can "launch" to the HPE Ezmeral Runtime Enterprise, landing to its dashboard. The kubectl, together with the HPE kubectl plugin and its kubeconfig, is all from this cross-launched dashboard. I will try to make this more clear in the Introduction page.
> 
> Guoping

Hi, Dale, with the new changes submitted, I believe all your comments have been addressed. 
In particularly, I rewrote a bit the Intro section by adding the fact that the HPE GreenLake for Containers being accessed and managed through HPE GreenLake Central portal and customers can 'cross launch' the HPE Ezmeral Runtime Enterprise to download the kubectl and the kubeconfig file in order to access the cluster in command line (common standard practice from K8s community).

I appreciate you can review it again. Just let me know in case anything I need to change. 

Thank you !

Guoping

drensing commented on content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md:
```diff
@@ -0,0 +1,261 @@
+
+2. Since launching from HPE GreenLake Central to the HPE Ezmeral Runtime Enterprise is configured through SAML SSO, a session token is fetched and added to the kubeconfig file each time you launch the HPE Ezmeral Runtime Enterprise dashboard. With HPE GreenLake for Containers, the session token is configured to expire after an hour. You will be unable to use the downloaded kubeconfig file to acce ...
+
+3. The kubeconfig file is generated to include some specific commands to show version, authenticate, and refresh the HPE Ezmeral Runtime Enterprise environment. The standard kubectl tool installed from the [official Kubernetes site](https://kubernetes.io/docs/tasks/tools/) does not work with this kubeconfig file. You have to download the `HPE kubectl plugin`, available from the same dashboard, an ...
```
In the last sentence, please modify "This causes issue due to..." to either "This causes an issue due to..." or "This causes issues due to..."

GuopingJia commented on content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md:
```diff
@@ -0,0 +1,261 @@
+  - hpe-greenlake
+---
+## Introduction
+[HPE GreenLake for Containers](https://www.hpe.com/us/en/greenlake/containers.html), one of the HPE GreenLake Cloud services, is built on  [HPE Ezmeral Runtime Enterprise](https://www.hpe.com/us/en/software/ezmeral-runtime.html) and deployed as an enterprise-grade container management service using open source Kubernetes. The HPE GreenLake for Containers service is accessed and managed through on ...
```
Very good comments and suggestions.

GuopingJia commented:
> 

Thanks Dale for your comments and good suggestions. They've all been addressed in the new changes.

GuopingJia commented:
Mathieu, 

As discussed this afternoon, I've added a paragraph in Conclusion to create a new service account and revoke the old one as the best practice for security. Let me know if you have any other comments.

Thank you !

Guoping

ma-los commented on content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md:
```diff
@@ -0,0 +1,267 @@
+
+3. The kubeconfig file is generated to include some specific commands to show version, authenticate, and refresh the HPE Ezmeral Runtime Enterprise environment. The standard kubectl tool installed from the [official Kubernetes site](https://kubernetes.io/docs/tasks/tools/) does not work with this kubeconfig file. You have to download the `HPE kubectl plugin`, available from the same dashboard, an ...
+
+You need a cleaner solution to have a kubeconfig file that is not tied to a specific user, has the right set of privileges, and works permanently with the standard kubectl tool. This blog post walks you through the process of creating a general-purpose kubeconfig file that allows you to access and deploy applications to the Kubernetes cluster in HPE GreenLake for Containers. By using these instru ...
```
can you reword "You need a cleaner solution to have a kubeconfig file that is not tied to a specific user, has the right set of privileges,"
You could introduce the fact you can free up from embeded ezmeral security enforcement and forge general-purpose kubeconfig bound to a service with specific roles and permissions which grant access to kubernetes cluster and application deployment in HPE GreenLake for Containers....

ma-los commented on content/blog/create-a-general-purpose-kubeconfig-file-in-hpe-greenlake-for-containers.md:
````diff
@@ -0,0 +1,267 @@
+```
+## Conclusion
+This blog post shows you how to create a general-purpose kubeconfig file using a service account. This kubeconfig file is not tied to any specific user. It is bound with a list of permissions carefully chosen for your need to access the Kubernetes cluster. The created kubeconfig file works permanently with both the downloaded kubectl binary from HPE GreenLake for Containers dashboard and the stan ...
+
````
Also you can point to the user doc : https://support.hpe.com/hpesc/public/docDisplay?docId=a00092451en_us&page=HPE-GreenLake-for-containers.html

GuopingJia commented:
Mathieu, 

Thank you for your time walking through together the blog contents yesterday. 
The new changes have addressed your comments (few comments seem to be outdated, they got addressed in the previous changes). 

Let me know if you have any further comments. 

Guoping
